### PR TITLE
qt: fix qt tools detection for Qt6.3+ by pkg-config

### LINKED ISSUE #10611: ArchLinux meson cannot detect Qt Tools with pkg-config.
**Describe the bug**
ArchLinux meson cannot detect Qt Tools with pkg-config.

**To Reproduce**
Run ./run_project_tests.py --only frameworks/4 on ArchLinux

**Expected behavior**
Qt 6 Tools should be detectable.

**system parameters**
*Native
*ArchLinux
* Python 3.10.5
* master
* ninja 1.11.0

### PR DESCRIPTION
**Pull request by rilian-la-te**
Fixes #10611

### CODE REVIEW

**Comment by eli-schwartz** [comment]
This really should not have anything to do with Arch Linux, it's a general transition in Qt6 which we fixed the generic way in the qmake handler and if this is relying on Arch-specific quirks then it needs to not be.

(For a long while the pkg-config detection would simply not be found, because Qt6 only recently re-added pkg-config support, sigh.)

**Comment by rilian-la-te** [comment]
@eli-schwartz fixed in generic way. This is also works in debian experimental too.

File: mesonbuild/dependencies/qt.py [hunk 1]
```diff
@@ -420,7 +427,8 @@ class Qt6PkgConfigDependency(QtPkgConfigDependency):
 
     @staticmethod
     def get_pkgconfig_host_bins(core: PkgConfigDependency) -> str:
-        return core.get_pkgconfig_variable('host_bins', [], None)
+        #Qt6 pkg-config for Qt defines bindir, and not host_bins
```
Comment (tristan957) [inline mesonbuild/dependencies/qt.py:430]: Put a space after the octothorpe (`#`)

**Comment by eli-schwartz** [comment]
Right so as I was concerned, this actually is not generic and does hardcode Arch Linux and I guess Debian too. It's not actually going to work correctly.

The solution for now is probably just use make...

**Comment by eli-schwartz** [comment]
> But the problem is than Qt6 pkg-config do not define libexecdir, but use it in build, so, we need to guess it like we will do in qt4 with bindir, I does not figured how to do it even more general way (path can vary).

Maybe for 6.3.2 we can use this: https://codereview.qt-project.org/c/qt/qtbase/+/422416

**Comment by rilian-la-te** [comment]
@eli-schwartz I created issue with a similar patch, will post link later today.

**Comment by eli-schwartz** [comment]
The patch is merged to dev and cherry-picked to at least 6.3 and 6.4, so we have a good solution lined up and I'd like to use that.

Hmm, when is the next scheduled point release for Qt?

**Comment by rilian-la-te** [comment]
@eli-schwartz will rewotk my PR this evening)

**Comment by eli-schwartz** [comment]
Already evening, huh?

**Comment by eli-schwartz** [comment]
Overall this looks good, though it would be nice to have confirmation it works.

One thing is that I'm not sure we should allow finding Qt6 via pkg-config for versions that don't have libexec defined, as the result will likely be broken in practice. We should fall back on qmake.

**Comment by eli-schwartz** [comment]
Also BTW, a couple days ago Qt6 landed (masked) in ::gentoo and the path is /usr/lib64/qt6/libexec so indeed the original version of this patch would not have worked there. :)

**Comment by eli-schwartz** [comment]
I added a commit that causes the pkg-config dependency to be reported as not-found, if libexecdir is not available. This should make it fall back to qmake in such a case.

**Review by thesamesam — APPROVED** [review]
This works for both broken Qt 6.3.1 (no libexecdir in .pc files) and patched Qt 6.3.1 (applying https://codereview.qt-project.org/c/qt/qtbase/+/422416) on Gentoo, which was broken beforehand.

It falls back correctly to qmake if lrelease can't be detected for unpatched Qt as well, so older Qt 6 series should be fine.

Thank you!

**Comment by eli-schwartz** [comment]
CI is broken due to weird Ubuntu issues we're still trying to debug, plus one macOS flake which will sort itself out when rerun during merge. Merged.

**Pull request merged**